engine/apps/api/views/schedule.py

```diff
@@ -289,7 +291,11 @@ def next_shifts_per_user(self, request, pk):
         users = {u: None for u in schedule.related_users()}
         for e in events:
             user = e["users"][0]["pk"] if e["users"] else None
-            if user is not None and users.get(user) is None and e["end"] > now:
+            event_end = e["end"]
+            if not isinstance(event_end, datetime.datetime):
```

Comment:
Will this be still needed after https://github.com/grafana/oncall/pull/742 is merged and [`end` is always `datetime.datetime`](https://github.com/grafana/oncall/blob/e103ec9152f0945deab760bf944c9d8a11975389/engine/apps/schedules/models/on_call_schedule.py#L230)?

Reply:
This endpoint is available for any type of schedule, so it may still break depending on how we call it. Eventually, if we unify the backend representation we should be able to remove this check (there are a few other places where we could also clean this up later).

Reply:
Hmm, I thought if it uses `final_events` which always passes `all_day_datetime=True` we should be good without this check here?

```
events = schedule.final_events(user_tz, starting_date, days=30)
```

Reply:
Oh, I see what you mean, you are right. With the other PR already merged, will update here and review the other occurrences I took note of :+1: